extensions/blocks/anchor-fm/editor.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Internal dependencies
+ */
+import { name } from '.';
+import getJetpackExtensionAvailability from '../../shared/get-jetpack-extension-availability';
+import { dispatch } from '@wordpress/data';
+import { createBlock } from '@wordpress/blocks';
+
+/*
+ * Register the main "anchor-fm" extension.
+ */
+const isExtensionAvailable = getJetpackExtensionAvailability( name )?.available;
+if ( isExtensionAvailable ) {
+	// Load data from the backend.
+	const data = window.Jetpack_AnchorFm;
+	if ( typeof data === 'object' ) {
+		// Insert the link badge if needed.
+		const { track, podcast_id, episode_id } = data;
+		if ( typeof track === 'object' && podcast_id && episode_id ) {
+			window.onload = () => {
+				const { link, spotify_show_url: spotifyShowUrl } = data;
+
+				// Insert badge.
+				dispatch( 'core/block-editor' ).insertBlocks(
```

Comment:
It may be worth checking the post meta is not already set to avoid duplicated badges.